src/components/forms/dns/DnsRecordForm.js

```diff
@@ -0,0 +1,250 @@
+import PropTypes from "prop-types";
+import { IcseFormGroup } from "../../Utils";
+import { IcseNameInput, IcseTextInput } from "../../Inputs";
+import { IcseSelect } from "../../Dropdowns";
+import { NumberInput } from "@carbon/react";
+import { isRangeInvalid } from "../../../lib/iam-utils";
+import { checkNullorEmptyString } from "../../../lib";
+
+/**
+ * DnsRecordForm
+ * @param {Object} props
+ */
+class DnsRecordForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...this.props.data };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    buildFormDefaultInputMethods(this);
+    buildFormFunctions(this);
+  }
+
+  /**
+   * handle input change
+   * @param {string} name key to change in state
+   * @param {*} value value to update
+   */
+  handleInputChange(event) {
+    this.setState(this.eventTargetToNameAndValue(event));
+  }
+
+  render() {
+    let dnsComponent = this.props.isModal
+      ? "new-dns-record"
+      : this.props.data.name;
+    return (
+      <>
+        <IcseFormGroup>
+          <IcseNameInput
+            id={this.state.name + "-name"}
+            labelText="DNS Record Name"
+            componentName={dnsComponent}
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            invalidCallback={() =>
+              this.props.invalidCallback(this.state, this.props)
+            }
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+            helperTextCallback={() =>
+              this.props.helperTextCallback(this.state, this.props)
+            }
+            className="fieldWidthSmaller"
+          />
+          <IcseSelect
+            name="dns_zone"
+            formName={dnsComponent + "-dns-zone"}
+            labelText="DNS Zone"
+            groups={this.props.dnsZones}
+            value={this.state.dns_zone}
+            handleInputChange={this.handleInputChange}
+            invalidText="Select a DNS Zone."
+            className="fieldWidthSmaller"
+          />
+          <IcseSelect
+            name="type"
+            formName={dnsComponent + "-type"}
+            labelText="DNS Record Type"
+            groups={["A", "AAA", "CNAME", "PTR", "TXT", "MX", "SRV"]}
+            value={this.state.type}
+            handleInputChange={this.handleInputChange}
+            invalidText="Select a DNS Record Type."
+            className="fieldWidthSmaller"
+          />
+        </IcseFormGroup>
+        <IcseFormGroup>
+          <IcseTextInput
+            componentName="DNS Record"
+            field="rdata"
+            labelText="Resource Data"
+            value={this.state.rdata}
+            id={this.state.name + "-rdata"}
+            onChange={this.handleInputChange}
+            invalid={this.props.invalidRdata(this.state, this.props)}
+            invalidText={this.props.invalidRdataText(this.state, this.props)}
+            className="fieldWidthSmaller"
```

Comment:
do we know how long the input for resource data can be? should this have the textInputWide class instead of fieldWidthSmaller?

Reply:
no idea, seems to be at best kind of random